apps/hexagon_histogram/pipeline.cpp

```diff
@@ -0,0 +1,62 @@
+#include "Halide.h"
+#include "process.h"
+using namespace Halide;
+
+// To use scatter-gathers add hvx_v65, hvx_scatter and hvx_gather
+// features to the target. Set HL_VTCM_SIZE environment value
+// to appropriate size (example 65536 -> 64KB)
+class Histogram : public Halide::Generator<Histogram> {
+public:
+
+    Input<Buffer<uint16_t>> input{"input", 1};
+    Output<Buffer<HIST_TYPE>> output{"output", 1};
+
+    void generate() {
+        Expr img_size = input.dim(0).extent();
+
+        Func histogram("histogram");
+        histogram(x) = cast<HIST_TYPE>(0);
+        RDom r(0, img_size);
+
+        Expr idx = clamp(input(r.x), 0, cast<uint16_t>(HIST_SIZE-1));
+        histogram(idx) += cast<HIST_TYPE>(1);
+        // Wrapper for the output
+        output(x) = histogram(x);
+
+        const int vector_size = get_target().has_feature(Target::HVX_128) ? 128 : 64;
+
+        histogram.vectorize(x, vector_size);
```

Comment:
Put all scheduling in one place or the other (not both generate() and schedule()).